### PR TITLE
KAA-1178: Administration guide > Cluster setup

### PR DESCRIPTION
**Pull request by krupet**

### CODE REVIEW

File: doc/Administration-guide/System-installation/Cluster-setup/index.md [hunk 1]
```diff
@@ -151,95 +149,417 @@ And for ```node3```
...
-node_list=<cassandra_database_ip>:<cassandra_database_port>
+  sudo apt-get install rsync
 ```
 
-Assuming that we peek standard Cassandra port, for all three nodes property would look like this
+### Configuring MariaDB Cluster
+
+#### MySQL Settings
+
+First of all, open the `"/etc/mysql/my.cnf"` file and comment the following lines **on all the three nodes**:
```
Comment (krupet) [inline doc/Administration-guide/System-installation/Cluster-setup/index.md]: > comment

make it bold

File: doc/Administration-guide/System-installation/Cluster-setup/MariaDB-cluster-setup-guide/index.md [hunk 2]
```diff
@@ -0,0 +1,270 @@
+---
+layout: page
+title: MariaDB cluster setup guide
+permalink: /:path/
+sort_idx: 30
+---
+
+* TOC
+{:toc}
+
```
Comment (balldir) [inline doc/Administration-guide/System-installation/Cluster-setup/MariaDB-cluster-setup-guide/index.md]: According to https://github.com/kaaproject/kaa/pull/842/files

> Right below the TOC, include a short introduction explaining the purpose of the page and what the reader can learn from it.

File: doc/Administration-guide/System-installation/Cluster-setup/MariaDB-cluster-setup-guide/index.md [hunk 3]
```diff
@@ -0,0 +1,270 @@
...
+   In the cluster configuration, each node that comes online tries to connect to at least one other node specified in its configuration file to get its initial state. 
+   Without the `--wsrep-new-cluster` parameter, this command would fail because the first node is unable to connect to any other nodes.
+
+3. On each of the other nodes, start MariaDB as you normally would.
+
+   ```bash
+     sudo service mysql start
+   ```
+
```
Comment (balldir) [inline doc/Administration-guide/System-installation/Cluster-setup/MariaDB-cluster-setup-guide/index.md:247]: Maybe some command to check cluster status ?

Comment (krupet) [inline doc/Administration-guide/System-installation/Cluster-setup/MariaDB-cluster-setup-guide/index.md:247]: Such commands listed right below.

**Comment by SerhiiKhomych** [comment]
LGTM

**Comment by SerhiiKhomych** [comment]
LGTM

File: doc/Administration-guide/System-installation/Cluster-setup/index.md [hunk 4]
```diff
@@ -5,62 +5,113 @@ permalink: /:path/
...
+
+## Requirements
+
+In order to set up Kaa cluster you need to have at least 3 Linux nodes with ```kaa-node``` service installed on each of them to create a reliable cluster, refer to [Single node installation]({{root_url}}Administration-guide/System-installation/Single-node-installation/) guide for installation details. 
 
-This page describes how to setup and configure kaa-node cluster.
+Also ```kaa-node``` service require some third party dependencies like SQL and NoSQL databases and Apache Zookeeper service, more details you can find in [Architecture overview]({{root_url}}Architecture-overview/). 
+You can find detailed instructions on how to install and configure Zookeeper service and one of supported SQL and NoSQL databases in [Single node installation]({{root_url}}Administration-guide/System-installation/Single-node-installation/#installation-steps) guide. 
+A set of databases (for example MongoDB + PostgreSQL) depends on your particular use case and some useful recommendations you can find in [Planning your deployment](http://10.2.2.133:4000/kaa/current/Administration-guide/System-installation/Planning-your-deployment/) guide.
```
Comment (Polinaa) [inline doc/Administration-guide/System-installation/Cluster-setup/index.md]: Change the link http://10.2.2.133:4000/...

File: doc/Administration-guide/System-installation/Cluster-setup/index.md [hunk 5]
```diff
@@ -5,62 +5,113 @@ permalink: /:path/
...
+{:toc}
 
-## Introduction
+This guide describes configuration of Kaa cluster on Linux nodes.
+
+The guide contains instructions on how to configure kaa-node service and required third party components like Apache Zookeeper service, SQL and NoSQL databases.
+
+## Requirements
+
+In order to set up Kaa cluster you need to have at least 3 Linux nodes with ```kaa-node``` service installed on each of them to create a reliable cluster, refer to [Single node installation]({{root_url}}Administration-guide/System-installation/Single-node-installation/) guide for installation details.
```
Comment (Polinaa) [inline doc/Administration-guide/System-installation/Cluster-setup/index.md:23]: All links to Kaa documentation return _Page not found_

File: doc/Administration-guide/System-installation/Cluster-setup/index.md [hunk 6]
```diff
@@ -150,83 +202,145 @@ And for ```node3```
...
+   ```bash
+    # specify jdbc database url
+    jdbc_url=jdbc:postgresql://172.1.1.1:5432,172.2.2.2:5432,172.3.3.3:5432/kaa
+   ```
+   
+   <br>
+   
+   </div></div>
+
+And also configure [username and password]({{root_url}}Administration-guide/System-installation/Single-node-installation#sql-database-configuration)
```
Comment (Polinaa) [inline doc/Administration-guide/System-installation/Cluster-setup/index.md]: Missing dot.

File: doc/Administration-guide/System-installation/Cluster-setup/index.md [hunk 7]
```diff
@@ -150,83 +202,145 @@ And for ```node3```
...
+    # specify jdbc database hosts and ports
+    jdbc_host_port=172.1.1.1:5432,172.2.2.2:5432,172.3.3.3:5432
+    
+    # specify jdbc database provider name
+    sql_provider_name=postgresql
+   ```
+   
+   </div></div>
+
+2. and ```/etc/kaa-node/conf/admin-dao.properties```
```
Comment (Polinaa) [inline doc/Administration-guide/System-installation/Cluster-setup/index.md]: Can you begin the second step not from _and_?

File: doc/Administration-guide/System-installation/Cluster-setup/index.md [hunk 8]
```diff
@@ -150,83 +202,145 @@ And for ```node3```
...
+
+For all three nodes it would be like this
 
 ```bash
- # Specify node list
-node_list=<cassandra_database_ip>:<cassandra_database_port>
+ nosql_db_provider_name=cassandra
 ```
 
+Setup Cassandra host ip in ```/etc/kaa-node/conf/common-dao-cassandra.properties``` file.
```
Comment (Polinaa) [inline doc/Administration-guide/System-installation/Cluster-setup/index.md]: IP

File: doc/Administration-guide/System-installation/Cluster-setup/index.md [hunk 9]
```diff
@@ -237,8 +351,16 @@ After all configuration properties set up
 If everithing is configured properly Kaa Administrator will see next output on command:
```
Comment (Polinaa) [inline doc/Administration-guide/System-installation/Cluster-setup/index.md]: everything

File: doc/Administration-guide/System-installation/Cluster-setup/index.md [hunk 10]
```diff
@@ -150,95 +204,165 @@ And for ```node3```
...
+   
+   </div></div>
+
+And also configure [username and password]({{root_url}}Administration-guide/System-installation/Single-node-installation#sql-database-configuration).
 
 ### NoSQL database configuration
 
-Select NoSQL database ```mongo``` or ```cassandra```in ```/etc/kaa-node/conf/dao.properties``` file.
+NoSQL database configuration will be almost the same as in single node setup too except for database node list, refer to [Single node installation guide - NoSQL database configuration]({{root_url}}Administration-guide/System-installation/Single-node-installation/#nosql-database-configuration) section for more details. 
+Select NoSQL database ```mongo``` or ```cassandra``` in ```/etc/kaa-node/conf/dao.properties``` file.
```
Comment (Polinaa) [inline doc/Administration-guide/System-installation/Cluster-setup/index.md]: _dao.properties_ -  No such file

File: doc/Administration-guide/System-installation/Cluster-setup/index.md [hunk 11]
```diff
@@ -150,95 +204,165 @@ And for ```node3```
...
+   
+   <div id="MariaDB1" class="tab-pane fade in active" markdown="1">
+   
+   <br>
+   
+   For all three nodes it would be like this
+   
+   ```bash
+   # specify jdbc database url
+   jdbc_url=jdbc:mysql:failover://172.1.1.1:5432,172.2.2.2:5432,172.3.3.3:5432/kaa
```
Comment (Polinaa) [inline doc/Administration-guide/System-installation/Cluster-setup/index.md]: Port 5432 is for PostgreSQL

File: doc/Administration-guide/System-installation/Cluster-setup/index.md [hunk 12]
```diff
@@ -150,95 +204,165 @@ And for ```node3```
...
+ nosql_db_provider_name=cassandra
 ```
 
+Setup Cassandra host IP in ```/etc/kaa-node/conf/common-dao-cassandra.properties``` file. 
 Assuming that we peek standard Cassandra port, for all three nodes property would look like this
 
 ```bash
- servers=172.1.1.1:9042
+ # Specify node list
+ node_list=127.1.1.1:9042,127.2.2.2:9042,127.3.3.3:9042
```
Comment (Polinaa) [inline doc/Administration-guide/System-installation/Cluster-setup/index.md]: **172**.1.1.1:9042,**172**.2.2.2:9042,**172**.3.3.3:9042

File: doc/Administration-guide/System-installation/Cluster-setup/index.md [hunk 13]
```diff
@@ -150,95 +204,165 @@ And for ```node3```
...
+
+   <div class="tab-content">
+   
+   <div id="MariaDB" class="tab-pane fade in active" markdown="1">
+
+   Configurations for all three nodes would look like this
+   
+   ```bash
+   # specify jdbc database hosts and ports
+   jdbc_host_port=127.1.1.1:3306,127.2.2.2:3306,127.3.3.3:3306
```
Comment (Polinaa) [inline doc/Administration-guide/System-installation/Cluster-setup/index.md]: **172**.1.1.1:3306,**172**.2.2.2:3306,**172**.3.3.3:3306

File: doc/Administration-guide/System-installation/Cluster-setup/MariaDB-cluster-setup-guide/index.md (hunk 3 above)
Comment (ashvayka) [inline doc/Administration-guide/System-installation/Cluster-setup/MariaDB-cluster-setup-guide/index.md:247]: There is much more simple commands: "SHOW STATUS LIKE 'wsrep_cluster_size';"
Please add a note somewhere on the page that one need to open following ports between this two nodes:
4444, 4567. This is important for wsrep communication.

File: doc/Administration-guide/System-installation/Cluster-setup/index.md [hunk 14]
```diff
@@ -5,62 +5,115 @@ permalink: /:path/
...
 
-In general cluster setup is similar to [single node setup](../Single-node-installation), except few details. We need at least 3 nodes to create a reliable cluster. For simplicity in this tutorial we will be using single instance of SQL and NoSQL databases, so this tutorial doesn't cover such themes like setting up Cassandra, MongoDB or PostgreSQL cluster setup. Let us assume that we have 3 nodes with Ubuntu 14.04 installed on each of them.
+In this guide we assume that you had already set up your SQL and NoSQL database clusters, so this tutorial doesn't cover such themes like setting up ones for Cassandra, MongoDB or PostgreSQL. 
+Refer to official [Cassandra](http://docs.datastax.com/en/landing_page/doc/landing_page/current.html), [MongoDB](https://docs.mongodb.com/manual/) and [PostgreSQL](https://www.postgresql.org/docs/) documentation in order to setup corresponding database cluster.
+
+MariaDB come with out of the box master-master replication support thus we recommend to use this database in your Kaa cluster.
```
Comment (ashvayka) [inline doc/Administration-guide/System-installation/Cluster-setup/index.md]: "with out of the box" need to be replaces with "with out-of-the-box" to avoid confusion during reading.

File: doc/Administration-guide/System-installation/Cluster-setup/index.md [hunk 15]
```diff
@@ -5,62 +5,115 @@ permalink: /:path/
...
+
+In addition to added on ```kaa-node``` service installation step firewall rules on every endpoint Kaa Administrator need to add some more rules for databases (depending on a set databases in the cluster) and Zookeeper ports.
 
 ```bash
-node1 172.1.1.1
-node2 172.2.2.2 
-node3 172.3.3.3
-```
+# MongoDB port
+$ sudo iptables -I INPUT -p tcp -m tcp --dport 27017 -j ACCEPT
```
Comment (ashvayka) [inline doc/Administration-guide/System-installation/Cluster-setup/index.md:65]: This settings are very specific. They will not affect AWS deployment or other providers. Please mention that this instructions depend on cloud provider.

File: doc/Administration-guide/System-installation/Cluster-setup/index.md [hunk 16]
```diff
@@ -5,62 +5,115 @@ permalink: /:path/
...
+ | sql_host_port             | 172.1.1.1:3306                                                                 | SQL database (MariaDB or PostgreSQL) host/ip_address and port    | /etc/kaa-node/conf/sql-dao.properties                                      |
+ | sql_provider_name         | postgresql, mysql:failover                                                     | JDBC database provider name                                      | /etc/kaa-node/conf/sql-dao.properties                                      |
+ | jdbc_url                  | jdbc:mysql:failover://172.1.1.1:3306/kaa, jdbc:postgresql://172.1.1.1:5432/kaa | SQL database host                                                | /etc/kaa-node/conf/admin-dao.properties                                    |
+ | nosql_db_provider_name    | cassandra/mongodb                                                              | Determines whether Cassandra or MongoDB provider will be used    | /etc/kaa-node/conf/nosql-dao.properties                                    |
```
Comment (ashvayka) [inline doc/Administration-guide/System-installation/Cluster-setup/index.md]: This is not related to cluster deploy, we may probably remove this.

File: doc/Administration-guide/System-installation/Cluster-setup/index.md (hunk 5 above)
Comment (krupet) [inline doc/Administration-guide/System-installation/Cluster-setup/index.md:23]: fixed

File: doc/Administration-guide/System-installation/Cluster-setup/MariaDB-cluster-setup-guide/index.md (hunk 3 above)
Comment (krupet) [inline doc/Administration-guide/System-installation/Cluster-setup/MariaDB-cluster-setup-guide/index.md:247]: done

**Comment by ashvayka** [comment]
Please add an important notice that each server should have different RSA keypairs. This is an important requirement and may cause weird bugs in production

File: doc/Administration-guide/System-installation/Cluster-setup/MariaDB-cluster-setup-guide/index.md [hunk 17]
```diff
@@ -0,0 +1,271 @@
...
+  wsrep_cluster_name="MariaDB_Cluster"
+  wsrep_node_address="ubuntu-node1"
+  wsrep_node_name="ubuntu-node1"
+  wsrep_sst_auth="root:'your password'"
+  wsrep_node_incoming_address=172.1.1.1
+  wsrep_sst_receive_address=172.1.1.1
+  wsrep_slave_threads=16
+```
+
+##### Configurations For `ubuntu-node2`
```
Comment (Polinaa) [inline doc/Administration-guide/System-installation/Cluster-setup/MariaDB-cluster-setup-guide/index.md]: Use **for** not from the capital letter

File: doc/Administration-guide/System-installation/Cluster-setup/MariaDB-cluster-setup-guide/index.md [hunk 18]
```diff
@@ -0,0 +1,271 @@
...
+  wsrep_cluster_name="MariaDB_Cluster"
+  wsrep_node_address="ubuntu-node2"
+  wsrep_node_name="ubuntu-node2"
+  wsrep_sst_auth="root:'your password'"
+  wsrep_node_incoming_address=172.2.2.2
+  wsrep_sst_receive_address=172.2.2.2
+  wsrep_slave_threads=16
+```
+
+##### Configurations For `ubuntu-node3`
```
Comment (Polinaa) [inline doc/Administration-guide/System-installation/Cluster-setup/MariaDB-cluster-setup-guide/index.md]: **for**

File: doc/Administration-guide/System-installation/Cluster-setup/MariaDB-cluster-setup-guide/index.md [hunk 19]
```diff
@@ -0,0 +1,271 @@
...
+2. Add the key files for the MariaDB repository.
+
+   ```bash
+     sudo apt-key adv --recv-keys --keyserver hkp://keyserver.ubuntu.com:80 0xcbcb082a1bb943db
+   ```
+
+3. Add the repository.
+
+   ```bash
+     sudo add-apt-repository 'deb http://mirror.jmu.edu/pub/mariadb/repo/5.5/ubuntu precise main'
```
Comment (Polinaa) [inline doc/Administration-guide/System-installation/Cluster-setup/MariaDB-cluster-setup-guide/index.md:53]: Should it be **trusty** for Ubuntu 14.04?

File: doc/Administration-guide/System-installation/Cluster-setup/MariaDB-cluster-setup-guide/index.md [hunk 20]
```diff
@@ -0,0 +1,271 @@
...
+
+Proceed to set the wsrep configurations on each node under the `[mysqld]`, using the specific hostname, root password and IP address of each node.
+
+##### Configurations for `ubuntu-node1`
+
+```bash
+  [mysqld]
+  wsrep_provider=/usr/lib/galera/libgalera_smm.so
+  wsrep_provider_options="gcache.size=256M; gcache.page_size=128M"
+  wsrep_cluster_address=gcomm://ubuntu-node1
```
Comment (Polinaa) [inline doc/Administration-guide/System-installation/Cluster-setup/MariaDB-cluster-setup-guide/index.md]: Should there be a list of IP addresses of cluster nodes?

File: doc/Administration-guide/System-installation/Cluster-setup/index.md [hunk 21]
```diff
@@ -5,62 +5,97 @@ permalink: /:path/
...
 
-## Cluster configuration
+```bash
+node1 172.1.1.1
+node2 172.2.2.2
+node3 172.3.3.3
+```
 
-### List of properties
+On each of nodes were installed NoSQl database (MongoDB or Cassandra), Zookeeper service and Kaa node service and everything that left it is to configure ```kaa-node``` services.
```
Comment (Polinaa) [inline doc/Administration-guide/System-installation/Cluster-setup/index.md]: NoSQL

File: doc/Administration-guide/System-installation/Cluster-setup/MariaDB-cluster-setup-guide/index.md [hunk 22]
```diff
@@ -0,0 +1,271 @@
...
+  wsrep_cluster_name="MariaDB_Cluster"
+  wsrep_node_address="ubuntu-node1"
```
Comment (Polinaa) [inline doc/Administration-guide/System-installation/Cluster-setup/MariaDB-cluster-setup-guide/index.md:121]: Should this parameter set the IP address for the individual node?

File: doc/Administration-guide/System-installation/Cluster-setup/index.md [hunk 23]
```diff
@@ -150,95 +186,165 @@ And for ```node3```
...
+   
+   </div></div>
+
+And also configure [username and password]({{root_url}}Administration-guide/System-installation/Single-node-installation#sql-database-configuration).
 
 ### NoSQL database configuration
 
-Select NoSQL database ```mongo``` or ```cassandra```in ```/etc/kaa-node/conf/dao.properties``` file.
+NoSQL database configuration will be almost the same as in single node setup too except for database node list, refer to [Single node installation guide - NoSQL database configuration]({{root_url}}Administration-guide/System-installation/Single-node-installation/#nosql-database-configuration) section for more details. 
+Select NoSQL database ```mongo``` or ```cassandra``` in ```/etc/kaa-node/conf/sql-dao.properties``` file.
```
Comment (Polinaa) [inline doc/Administration-guide/System-installation/Cluster-setup/index.md]: nosql-dao.properties

File: doc/Administration-guide/System-installation/Cluster-setup/index.md [hunk 24]
```diff
@@ -5,62 +5,97 @@ permalink: /:path/
...
+# PostgreSQL port
+$ sudo iptables -I INPUT -p tcp -m tcp --dport 5432 -j ACCEPT
+$ sudo iptables -I OUTPUT -p tcp -m tcp --dport 5432 -j ACCEPT
+
+# Zookeeper port
+$ sudo iptables -I INPUT -p tcp -m tcp --dport 2181 -j ACCEPT
+$ sudo iptables -I OUTPUT -p tcp -m tcp --dport 2181 -j ACCEPT
+
+$ sudo apt-get install iptables-persistent
+$ sudo service netfilter-persistent start
```
Comment (Polinaa) [inline doc/Administration-guide/System-installation/Cluster-setup/index.md]: _unrecognized service_ for Ubuntu 14.04

File: doc/Administration-guide/System-installation/Cluster-setup/MariaDB-cluster-setup-guide/index.md (hunk 19 above)
Comment (krupet) [inline doc/Administration-guide/System-installation/Cluster-setup/MariaDB-cluster-setup-guide/index.md:53]: Guide was verified with _precise_ on Ubuntu 14.04.

**Comment by Polinaa** [comment]
To allow remote connections I change _bind_ip_ in _/etc/mongod.conf_ and _bind-address_ in _/etc/mysql/my.cnf_.

**Comment by krupet** [comment]
There is a note in the documentation that you need to allow external connections in DB's security configuration if DB is set up on another host.

File: doc/Administration-guide/System-installation/Cluster-setup/MariaDB-cluster-setup-guide/index.md [hunk 25]
```diff
@@ -0,0 +1,275 @@
...
+We need at least 3 hosts running together with Ubuntu 14.04 Operating system to form a reliable cluster. 
+The following is the hosts list that we had setup for this article, where we will deploy the MariaDB Galera cluster:
+
+```bash
+  ubuntu-node1 172.1.1.1 
+  ubuntu-node2 172.2.2.2 
+  ubuntu-node3 172.3.3.3
+```
+
+Now we will install the its required packages `rsync`, `galera` and `mariadb-galera-server` that need to be installed on all the three nodes.
```
Comment (Polinaa) [inline doc/Administration-guide/System-installation/Cluster-setup/MariaDB-cluster-setup-guide/index.md]: the required

File: doc/Administration-guide/System-installation/Cluster-setup/index.md [hunk 26]
```diff
@@ -5,62 +5,121 @@ permalink: /:path/
...
+> 
+> $ sudo md5sum /usr/lib/kaa-node/keys/operations/public.key 
+>  8d9b8d4838ff03ddac52dec97797546c /usr/lib/kaa-node/keys/operations/public.key
+> 
+> $ sudo md5sum /usr/lib/kaa-node/keys/operations/private.key 
+>  f9aab71ef879916d24dfd5200ad94ccd /usr/lib/kaa-node/keys/operations/private.key 
+> ```
+> In order to refresh key files stop ```kaa-node``` service, remove key files and start the service again.
+
+Also ```kaa-node``` service require some third party dependencies like SQL and NoSQL databases and Apache Zookeeper service, more details you can find in [Architecture overview]({{root_url}}Architecture-overview/).
```
Comment (Polinaa) [inline doc/Administration-guide/System-installation/Cluster-setup/index.md]: requires

**Comment by Polinaa** [comment]
There is no reference to **Apache Zookeeper** in Cluster setup guide.

**Comment by Polinaa** [comment]
LGTM

File: doc/Administration-guide/System-installation/Cluster-setup/index.md [hunk 27]
```diff
@@ -5,62 +5,121 @@ permalink: /:path/
...
+# MariaDB port 
+$ sudo iptables -I INPUT -p tcp -m tcp --dport 3306 -j ACCEPT
+$ sudo iptables -I OUTPUT -p tcp -m tcp --dport 3306 -j ACCEPT
+
+# PostgreSQL port
+$ sudo iptables -I INPUT -p tcp -m tcp --dport 5432 -j ACCEPT
+$ sudo iptables -I OUTPUT -p tcp -m tcp --dport 5432 -j ACCEPT
+
+# Zookeeper port
+$ sudo iptables -I INPUT -p tcp -m tcp --dport 2181 -j ACCEPT
```
Comment (ashvayka) [inline doc/Administration-guide/System-installation/Cluster-setup/index.md:95]: 2888,3888 is required for ZK master discovery

File: doc/Administration-guide/System-installation/Cluster-setup/index.md [hunk 28]
```diff
@@ -5,62 +5,121 @@ permalink: /:path/
...
+# MongoDB port
+$ sudo iptables -I INPUT -p tcp -m tcp --dport 27017 -j ACCEPT
+$ sudo iptables -I OUTPUT -p tcp -m tcp --dport 27017 -j ACCEPT
+
+# Cassandra port
+$ sudo iptables -I INPUT -p tcp -m tcp --dport 9042 -j ACCEPT
+$ sudo iptables -I OUTPUT -p tcp -m tcp --dport 9042 -j ACCEPT
+
+# MariaDB port 
+$ sudo iptables -I INPUT -p tcp -m tcp --dport 3306 -j ACCEPT
```
Comment (ashvayka) [inline doc/Administration-guide/System-installation/Cluster-setup/index.md:83]: 4444 and 4567 is required for rpc between mariadb nodes

File: doc/Administration-guide/System-installation/Cluster-setup/index.md (hunk 27 above)
Comment (ashvayka) [inline doc/Administration-guide/System-installation/Cluster-setup/index.md:95]: 7000 port is also required, but i don't recall who is using it. close all ports and check who is failing

**Comment by ashvayka** [comment]
Few minor comments about ports, otherwise - LGTM

File: doc/Administration-guide/System-installation/Cluster-setup/index.md (hunk 27 above)
Comment (krupet) [inline doc/Administration-guide/System-installation/Cluster-setup/index.md:95]: done.

**Comment by ashvayka** [comment]
LGTM

File: doc/Administration-guide/System-installation/Cluster-setup/index.md [hunk 29]
```diff
@@ -5,62 +5,139 @@ permalink: /:path/
...
+$ sudo iptables -I INPUT -p tcp -m tcp --dport 7199 -j ACCEPT
+$ sudo iptables -I OUTPUT -p tcp -m tcp --dport 7199 -j ACCEPT
+# for the Cassandra Thrift client
+$ sudo iptables -I INPUT -p tcp -m tcp --dport 9160 -j ACCEPT
+$ sudo iptables -I OUTPUT -p tcp -m tcp --dport 9160 -j ACCEPT
+
+# MariaDB port 
+$ sudo iptables -I INPUT -p tcp -m tcp --dport 3306 -j ACCEPT
+$ sudo iptables -I OUTPUT -p tcp -m tcp --dport 3306 -j ACCEPT
+$ sudo iptables -I INPUT -p tcp -m tcp --dport 2888 -j ACCEPT
```
Comment (Polinaa) [inline doc/Administration-guide/System-installation/Cluster-setup/index.md:85]: 2888,3888 is required for ZK.
4444 and 4567 - for mariadb.
